internal/api/server.go

```diff
@@ -2,107 +2,220 @@ package api
 
 import (
 	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
 	"log"
 
 	"github.com/google/uuid"
-	"github.com/rhobs/rhobs-synthetics-api/pkg/apis/v1"
+	v1 "github.com/rhobs/rhobs-synthetics-api/pkg/apis/v1"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/kubernetes"
 )
 
-type Server struct{}
+type Server struct {
+	KubeClient kubernetes.Interface
+	Namespace  string
+}
 
-func NewServer() Server {
-	return Server{}
+func NewServer(client kubernetes.Interface, namespace string) Server {
+	return Server{
+		KubeClient: client,
+		Namespace:  namespace,
+	}
 }
 
 // (GET /metrics/probes)
-func (Server) ListProbes(ctx context.Context, request v1.ListProbesRequestObject) (v1.ListProbesResponseObject, error) {
-	// Fake respone while we wire things together
-	clusterId, err := uuid.Parse("957c5277-f74c-4b24-938a-f70bab28aab5")
-	if err != nil {
-		return v1.ListProbes400JSONResponse{
-			Error: struct {
-				Code    int32  `json:"code"`
-				Message string `json:"message"`
-			}{
-				Code:    400,
-				Message: "Invalid Cluster ID",
-			},
-		}, nil
+func (s Server) ListProbes(ctx context.Context, request v1.ListProbesRequestObject) (v1.ListProbesResponseObject, error) {
+	// Start with the base selector for our app
+	baseSelector := "app=rhobs-synthetics"
+	finalSelector := baseSelector
+
+	// If the user provided a selector, validate and append it
+	if request.Params.LabelSelector != nil && *request.Params.LabelSelector != "" {
+		userSelector := *request.Params.LabelSelector
+		// Validate the user-provided selector syntax
+		_, err := labels.Parse(userSelector)
+		if err != nil {
+			return v1.ListProbes400JSONResponse{
+				Error: &struct {
+					Code    *int32  `json:"code,omitempty"`
+					Message *string `json:"message,omitempty"`
+				}{
+					Code:    int32p(400),
+					Message: stringp(fmt.Sprintf("invalid label_selector: %v", err)),
+				},
```

Comment:
This syntax is throwing me: don't we already define the error struct in our `api/v1` pkg? If so, couldn't we just return the following?

```suggestion
				{
					Code:    int32p(400),
					Message: stringp(fmt.Sprintf("invalid label_selector: %v", err)),
				},
```

Reply:
Additionally, why can we set a different error code than `400` for a struct named `ListProbes400JSONResponse`?

Reply:
RE: different error code:

400 is for a bad request made by the client (aka the client messed up the label selector, something like`private===true`.
https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Status/400 

Did you have thoughts on a different code?